### ISSUE TITLE
Directly calling "AbortController" when checking for support is not supported on older versions of browsers

### ISSUE THREAD

**Issue by DotIN13** [maplibre/maplibre-gl-js#1424]
**maplibre-gl-js version**:
2.1.9

**browser**:
Chromium Version 65.0.3325.181 (Official Build) Built on Ubuntu , running on LinuxMint 21 (64-bit)

### Steps to Trigger Behavior

 1. Build with webpack and babel, transpiled to es5 with @babel/preset
 2. Open a webpage with any maplibre map on Chromium version less than 66
 3. The map would not render

I assume it would be great to be able to run maplibre on older browsers after transpiling, even if we are moving forward to es6.

### Minimal Example

[CodeSandbox](https://codesandbox.io/s/blue-violet-vjuuj5?file=/src/index.js)
Chromium downloaded from [Ubuntu archives](http://archive.ubuntu.com/ubuntu/pool/universe/c/chromium-browser/?C=M;O=D)

### Expected Behavior

The makeRequest() in `/src/utils/ajax.ts` runs, and map renders.

### Actual Behavior

```javascript
if (fetch && Request && AbortController && Object.prototype.hasOwnProperty.call(Request.prototype, 'signal')) {
    return makeFetchRequest(requestParameters, callback);
}
```

Calling [AbortController](https://github.com/maplibre/maplibre-gl-js/blob/15b24245d734dd0a79e88b870ea54fe451ef61ac/src/util/ajax.ts#L285) will result in an error as follows as the standard is not implemented until Chromium 66.

```log
Uncaught ReferenceError: AbortController is not defined
    at K (:36207)
    at Object.__webpack_modules__../.yarn/unplugged/maplibre-gl-npm-2.1.9-5f019a0db3/node_modules/maplibre-gl/dist/maplibre-gl.js.t.getJSON (:36207)
    at ie.loadURL (:36211)
    at Map._updateStyle (:36211)
    at Map.setStyle (:36211)
    at new Map (:36211)
```

### Possible solution

If interest in backward compatibility still exists, it is possible to add a `try catch` block around the support check, and when an error is catched, fallback to XHRs.

```javascript
try {
    if (fetch && Request && AbortController && Object.prototype.hasOwnProperty.call(Request.prototype, 'signal')) {
        return makeFetchRequest(requestParameters, callback);
    }
} catch (err) {
    console.log('AbortController is not supported, use XHR instead.');
}
```

**Comment by HarelM** [comment]
We do not support older browsers, much like we do not support IE in latest versions - they have security issue and backwards compatibility issues.
You can use polyfills to solve this I believe.

**Comment by DotIN13** [comment]
I see, I am currently using yarn patches and npm patches to run maplibre on older browsers. It's kinda messy, but a lot of my users needed that compatibility, so...

Overall I understand the concerns and thanks anyways.

**Comment by acalcutt** [comment]
I've made an attempt to maintain old browser support at https://github.com/acalcutt/maplibre-gl-js/tree/2.x_ie_compat . It "works" for the most part but it seems pretty unstable when zooming into the details. I'd love to figure out the why of the instability. Its seemed to start around the Time @birkskyum started trying to modernize stuff...before that it was working pretty well 

For the AbortController issue you mentioned I added these packages to package.json
```
    "abortcontroller-polyfill": "^1.7.3",
    "promise-polyfill": "8.2.1",
    "unfetch": "^4.2.0"
```
    
Then in /src/utils/ajax.ts I added 
```
import 'promise-polyfill/src/polyfill';
import 'unfetch/polyfill';
import 'abortcontroller-polyfill';
```

This in basically backdoored in AbortController support through polyfill.

**Comment by DotIN13** [comment]
@acalcutt I was not so desperate to have to support IE, but I did needed to provide support to some nasty old Chromium-based in my project.

The same as you did, I added the AbortController polyfill. Besides that I also had to workaround the @mapbox/tiny-sdf modernization since the advanced TextMetrics is not supported on older browsers. I had to dig up the code where they still had back-compatibility and apply a patch to the npm package manually.

So far so good, I have managed to get my own maplibre instance running on Chromium 61 (Compared to the released version's 77).
